### PR TITLE
Update snapshot to 7123292

### PR DESCRIPTION
**Pull request by ricknout**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -74,9 +74,20 @@ subprojects {
...
+            freeCompilerArgs += '-Xopt-in=kotlin.RequiresOptIn'
+        }
+    }
 
-            freeCompilerArgs += "-Xallow-jvm-ir-dependencies"
+    // FIXME: Only needed for Compose 1.0.0-alpha12 snapshots
+    configurations.configureEach {
+        resolutionStrategy.eachDependency { DependencyResolveDetails details ->
+            if (details.requested.group == 'androidx.lifecycle' &&
+                    details.requested.name != 'lifecycle-viewmodel-compose') {
```
Comment (chrisbanes) [inline build.gradle:85]: Only need this if you're using `viewModel()`

File: lib/src/main/java/com/google/android/material/composethemeadapter/TypedArrayUtils.kt [hunk 2]
```diff
@@ -157,18 +157,18 @@ internal fun parseShapeAppearance(
         when (a.getInt(R.styleable.ComposeThemeAdapterShapeAppearance_cornerFamily, 0)) {
             0 -> {
                 RoundedCornerShape(
-                    topLeft = cornerSizeTL ?: defaultCornerSize ?: fallbackShape.topLeft,
-                    topRight = cornerSizeTR ?: defaultCornerSize ?: fallbackShape.topRight,
-                    bottomRight = cornerSizeBR ?: defaultCornerSize ?: fallbackShape.bottomRight,
-                    bottomLeft = cornerSizeBL ?: defaultCornerSize ?: fallbackShape.bottomLeft
+                    topStart = cornerSizeTL ?: defaultCornerSize ?: fallbackShape.topStart,
```
Comment (chrisbanes) [inline lib/src/main/java/com/google/android/material/composethemeadapter/TypedArrayUtils.kt:160]: Probably need to convert left/right to start/end here. Fine for now, but should be fixed before alpha12

**Review by chrisbanes — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (ricknout) [reply build.gradle:85]: It seems to also be needed for `androidx.lifecycle:lifecycle-runtime` which appears to be a transitive dependency of MDC and AndroidX Core
```
Could not find androidx.lifecycle:lifecycle-runtime:2.3.0.
Required by:
    project :lib > com.google.android.material:material:1.1.0
    project :lib > androidx.core:core:1.5.0-alpha01
```

Comment (chrisbanes) [reply build.gradle:85]: Ahhhh, fair enough.

File: lib/src/main/java/com/google/android/material/composethemeadapter/TypedArrayUtils.kt (hunk 2 above)
Comment (ricknout) [reply lib/src/main/java/com/google/android/material/composethemeadapter/TypedArrayUtils.kt]: Done

File: lib/src/main/java/com/google/android/material/composethemeadapter/TypedArrayUtils.kt [hunk 3]
```diff
@@ -157,18 +157,18 @@ internal fun parseShapeAppearance(
         when (a.getInt(R.styleable.ComposeThemeAdapterShapeAppearance_cornerFamily, 0)) {
             0 -> {
                 RoundedCornerShape(
-                    topLeft = cornerSizeTL ?: defaultCornerSize ?: fallbackShape.topLeft,
-                    topRight = cornerSizeTR ?: defaultCornerSize ?: fallbackShape.topRight,
-                    bottomRight = cornerSizeBR ?: defaultCornerSize ?: fallbackShape.bottomRight,
-                    bottomLeft = cornerSizeBL ?: defaultCornerSize ?: fallbackShape.bottomLeft
+                    topStart = cornerSizeTS ?: defaultCornerSize ?: fallbackShape.topStart,
```
Comment (chrisbanes) [inline lib/src/main/java/com/google/android/material/composethemeadapter/TypedArrayUtils.kt:160]: Just checking, but this isn't what I meant. The attributes are left/right so you need to convert them to start/end based on the layout direction. Currently the values are only correct in LTR, not RTL.

Comment (ricknout) [reply lib/src/main/java/com/google/android/material/composethemeadapter/TypedArrayUtils.kt:160]: Aaah my bad. I'll open a new PR

Comment (ricknout) [reply lib/src/main/java/com/google/android/material/composethemeadapter/TypedArrayUtils.kt:160]: https://github.com/material-components/material-components-android-compose-theme-adapter/pull/44

**Pull request merged**